code/game/objects/effects/poster.dm

```diff
@@ -150,6 +152,8 @@
 
 /obj/structure/sign/poster/proc/randomise(base_type)
 	var/list/poster_types = subtypesof(base_type)
+	if(random_blacklisted_basetype)
+		poster_types -= typesof(random_blacklisted_basetype)
```

Comment:
keeping in with the list theme, this would be

```suggestion
	if(length(blacklisted_types))
		for(var/iterated_type in blacklisted_types)
		poster_types -= typesof(iterated_type)
```